### DISCUSSION TITLE
Docker or VMWare

### DISCUSSION THREAD

**Discussion by kavikode** [lucasw/rviz_camera_stream#33] [Q&A]
Hello, 

This is a great package and very useful for robotics! I am a beginner in ROS and Ubuntu is I challenge to use because of my severe physical disability.

I'm planning to use Docker or VMWare. Is there a docker image or instructions to install in VMWare available with your package?

It will be very helpful as I am a beginner and I truly appreciate.
Thank you

**Comment by lucasw** [comment]
Probably a virtual machine with vmware (or `qemu-kvm`) is going to be easier than docker if you are just starting out.  You'd go through installing Ubuntu from the iso, then do the regular ros ubuntu setup instructions, then create a catkin_ws and git clone this repo inside it, then could build it and run rviz inside it.

http://wiki.ros.org/docker/Tutorials/Docker has the basic instructions for getting ros running in a docker container.  Inside the container you would need a `catkin_ws/src` and get the source code to this repo by either running git clone inside it, or mapping it in from the outside, then install any additional dependencies not provided by the base docker ros image, then catkin build rviz_camera_stream. There are going to be additional hurdles getting rviz (or any graphical application) running inside docker though.

**Comment by kavikode** [comment]
Thank you @lucasw I appreciate your fast response
